### ISSUE TITLE
Improve timestamp flexibility for custom logging backends.

### ISSUE THREAD

**Issue by matharman** [zephyrproject-rtos/zephyr#41809]
There are a few small improvements I'd like to suggest (and contribute as a PR!) to the logging subsystem. Before I dive in to the changes, I wanted to make sure I fully understood any historical context or rationale for why these conditions exist to begin with.

1. Remove [symbol dependencies](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/logging/Kconfig.formatting#L85) on `CONFIG_LOG_BACKEND_xxx` from `CONFIG_LOG_BACKEND_FORMAT_TIMESTAMP`. This restricts out-of-tree, custom logging backends from enabling the formatted timestamp, _unless_ one of the dependency backends is also enabled. It seems to me that any log backend is able to process the flag `LOG_OUTPUT_FLAG_FORMAT_TIMESTAMP`, so I wonder if there is a reason for this requirement?
2. Enable log backends besides `LOG_BACKEND_NET`, or those which do not set `LOG_OUTPUT_FLAG_FORMAT_SYSLOG`, to use the UTC timestamp formatting code ([subsys/logging/log_output.c:200](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/logging/log_output.c#L200)). Instead, I suggest that the UTC timestamp formatting could be gated by something like a new Kconfig choice symbol selecting either the default format `hh:mm:ss:ms,us`, or UTC.
    - Alternatively, this might be better handled by a PR targeting https://github.com/zephyrproject-rtos/zephyr/issues/41146 which provides default formatters for UTC and `hh:mm:ss:ms,us`.

**Is your enhancement proposal related to a problem? Please describe.**

Suggestion (1) would allow out-of-tree custom logging backends to `imply CONFIG_LOG_BACKEND_FORMAT_TIMESTAMP` without managing a fork or patches to Zephyr. This is an experimental use case I have (not time-sensitive).

Suggestion (2) is related to my experimental use case as well, where UTC timestamps in the logs would be very helpful in some end-to-end testing. I think that enabling UTC timestamps for any logging would be a nice additional flexibility.

**Describe the solution you'd like**

I would like to contribute 2 PRs, one for each suggestion detailed above, if maintainers agree with my ideas. No worries if this a bit out of scope for a first time contribution, I would completely understand.

**Describe alternatives you've considered**
For (1):
- Maintaining a fork of Zephyr with patches supporting my use case, which I don't find desirable.

**Additional context**
N/A

**Comment by zephyrbot** [comment]
Hi @nordic-krch,

This issue, marked as an Enhancement, was opened a while ago and did not get any traction. It was just assigned to you based on the labels. If you don't consider yourself the right person to address this issue, please re-assing it to the right person.

Please take a moment to review if the issue is still relevant to the project.  If it is, please provide feedback and direction on how to move forward. If it is not, has already been addressed, is a duplicate, or is no longer relevant, please close it with a short comment explaining the reason.


@matharman you are also encouraged to help moving this issue forward by providing additional information and confirming this request/issue is still relevant to you.

Thanks!

**Comment by matharman** [comment]
Resolved by #56277
